### DISCUSSION TITLE
confused with bottle naming and generated bottle files

### DISCUSSION THREAD

**Discussion by ipatch** [Homebrew/discussions#1945] [Tap maintenance and Homebrew development]
i'm updating the [homebrew-freecad](https://github.com/FreeCAD/homebrew-freecad) tap to use bottles hosted on github via github releases. for macos mojave i'm building these bottles locally on my personal machine. i'm running into some issues unknown to me where the bottles being generated with the below commands have two hyphens in them ie. `--`

---

```
brew install --build-bottle freecad/freecad/tbb@2020_u3
brew bottle freecad/freecad/tbb@2020_u3
```

gives me the below files with the below names in my current directory,

```
tbb@2020_u3--2020_U3_1.mojave.bottle.1.tar.gz
tbb@2020_u3--2020_U3_1.mojave.bottle.json
```

then when i go to install `tbb@2020_u3` with `brew install tbb@2020_u3` the curl cmd in the `brew install` process is unable to locate the bottle file _tar.gz_ from the github releases page. it seems to be looking for a file name of

```
tbb@2020_u3-2020_U3_1.mojave.bottle.tar.gz
```

i have not a clue why the `brew install` process would be looking for `-` and not `--` nor do i understand why the `brew install --build-bottle` and `brew bottle` process/commands are creating filenames with `--`

any and old help would be greatful 🙏

thanks
chris

**Comment by gromgit** [comment]
This is an convention from years ago (https://github.com/Homebrew/brew/issues/4740), and the recommended practice still seems to be:
1. Use `brew test-bot` to handle build+bottle+upload, it's the supported way.
2. If you absolutely can't use `brew test-bot`, parse the bottle JSON for the `local_filename` and `filename` fields. The former is the name used on your filesystem, the latter for when you upload.

**Reply by ipatch** [reply]
thank you. i'll look into this tomorrow. as always your replies are welcome and insightful.
